src/user/user.service.ts

```diff
@@ -25,29 +29,48 @@ export class UserService {
     return createdUser.save();
   }
 
+  async registerUser(userData: CreateUserDto) {
+    try {
+      if (
+        !userData.email ||
+        !userData.password ||
+        !userData.name ||
+        !userData.lastname ||
+        !userData.phone
+      ) {
+        throw new HttpException(
+          MISSING_REQUIRED_FIELDS,
+          HttpStatus.UNPROCESSABLE_ENTITY,
+        );
+      }
+      const existingUser = await this.findUserByEmail(userData.email);
+      if (existingUser) {
+        throw new HttpException(USER_WITH_THIS_EMAIL, HttpStatus.CONFLICT);
+      }
+      return this.createUser({ ...userData, role: 'user' });
+    } catch (error) {
+      throw new HttpException(error.message, error.status);
```

Comment:
Write in more detail why this line was added, and how the status code will be determined